.travis.yml.nix

```diff
@@ -20,34 +22,44 @@ let
   scripts = {
     builds = {
       name = "nix-build";
-      script = ''
-        set -e
-        source ./.travis_fold.sh
-        lorri_travis_fold lorri-nix-build \
-          nix-build
-        lorri_travis_fold lorri-install \
-          nix-env -i ./result
-        lorri_travis_fold lorri-self-upgrade \
-          lorri self-upgrade local $(pwd)
-      '';
+      script = [
+        ''
+          set -e
+          source ./.travis_fold.sh
+          lorri_travis_fold lorri-nix-build \
+            nix-build
+          lorri_travis_fold lorri-install \
+            nix-env -i ./result
+          lorri_travis_fold lorri-self-upgrade \
+            lorri self-upgrade local $(pwd)
+        ''
+        # push build closure to cachix
+        ''readlink ./result >> ${cachix-queue-file}''
+      ];
     };
 
     lints = {
       name = "cargo build & linters";
-      script = ''
-        set -e
-        source ./.travis_fold.sh
-
-        lorri_travis_fold ci_check \
-          nix-shell --quiet --arg isDevelopmentShell false --run ci_check
-        lorri_travis_fold travis-yml-gen \
-          cat $(nix-build --quiet ./.travis.yml.nix --no-out-link) > .travis.yml
-        lorri_travis_fold travis-yml-idempotent \
-          git diff -q ./.travis.yml
-        lorri_travis_fold carnix-idempotent \
-          git diff -q ./Cargo.nix
+      script = [
+        ''
+          set -e
+          source ./.travis_fold.sh
 
-      '';
+          lorri_travis_fold ci_check \
+            nix-shell --quiet --arg isDevelopmentShell false --run ci_check
+          lorri_travis_fold travis-yml-gen \
+            cat $(nix-build --quiet ./.travis.yml.nix --no-out-link) > .travis.yml
+          lorri_travis_fold travis-yml-idempotent \
+            git diff -q ./.travis.yml
+          lorri_travis_fold carnix-idempotent \
+            git diff -q ./Cargo.nix
+        ''
+        # push test suite closure to cachix
+        ''
+          buildInputs=$(nix-build -E '(import ./shell.nix { isDevelopmentShell = false; }).buildInputs')
+          printf '%s' "$buildInputs" >> ${cachix-queue-file}
```

Comment:
why `printf` instead of just ` >> $HOME/push-to-cachix`?

Reply:
Ah, that’s an artifact from backporting from a later commit, will change.